### PR TITLE
[test-only] delete users test

### PR DESCRIPTION
**Pull request by ScharfViktor**
- [x] delete users using context menu
- [x] delete users using batch action

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/33710/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33710/tracing/copy-and-move-resources-in-personal-space-alice-2023-3-15-10-25-11.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33710/tracing/user-login-can-be-managed-in-the-admin-settings-admin-2023-3-15-10-23-44.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33710/tracing/user-login-can-be-managed-in-the-admin-settings-admin-2023-3-15-10-23-51.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/33710/tracing/user-login-can-be-managed-in-the-admin-settings-alice-2023-3-15-10-23-49.zip 


</p></details>
:boom: The e2e-ocis tests pipeline failed. The build has been cancelled.

File: tests/e2e/cucumber/steps/ui/adminSettings.ts [hunk 1]
```diff
@@ -300,3 +300,33 @@ When(
...
   }
 )
+
+When(
+  /^"([^"]*)" deletes the following (user|users) using the (batch actions|context menu)$/,
+  async function (
+    this: World,
+    stepUser: string,
+    _: string,
+    way: string,
```
Comment (SagarGi) [inline tests/e2e/cucumber/steps/ui/adminSettings.ts:310]:
```suggestion
    actionType: string,
```
how about using this?

File: tests/e2e/cucumber/steps/ui/adminSettings.ts [hunk 2]
```diff
@@ -300,3 +300,33 @@ When(
...
+    stepTable: DataTable
+  ): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const usersObject = new objects.applicationAdminSettings.Users({ page })
+
+    switch (way) {
```
Comment (SagarGi) [inline tests/e2e/cucumber/steps/ui/adminSettings.ts:316]:
```suggestion
    switch (actionType) {
```

File: tests/e2e/cucumber/steps/ui/adminSettings.ts [hunk 3]
```diff
@@ -300,3 +300,33 @@ When(
...
+
+When(
+  /^"([^"]*)" deletes the following (user|users) using the (batch actions|context menu)$/,
+  async function (
+    this: World,
+    stepUser: string,
+    _: string,
+    actionType: string,
+    stepTable: DataTable
+  ): Promise<void> {
```
Comment (saw-jan) [inline tests/e2e/cucumber/steps/ui/adminSettings.ts:312]: I think we can do this in regex
```suggestion
When(
  /^"([^"]*)" deletes the following (?:user|users) using the (batch actions|context menu)$/,
  async function (
    this: World,
    stepUser: string,
    actionType: string,
    stepTable: DataTable
  ): Promise<void> {
```

File: tests/e2e/cucumber/steps/ui/adminSettings.ts [hunk 4]
```diff
@@ -300,3 +300,33 @@ When(
...
+        }
+        await usersObject.deleteUserUsingBatchAction()
+        break
+      case 'context menu':
+        for (const { user } of stepTable.hashes()) {
+          await usersObject.deleteUserUsingContextMenu({ key: user })
+        }
+        break
+      default:
+        throw new Error(`${actionType} not implemented`)
```
Comment (saw-jan) [inline tests/e2e/cucumber/steps/ui/adminSettings.ts:329]:
```suggestion
        throw new Error(`'${actionType}' not implemented`)
```

**Review by saw-jan — COMMENTED** [review]
LGTM

**Pull request merged**